docs/build/compute/architecture.md

```diff
@@ -0,0 +1,107 @@
+
+#### Core Functions
+
+- **Docker Compose Management**: Launches and manages the user's containerized workload
+- **Log Collection**: Provides access to both cvm-agent logs and container logs
+- **System Monitoring**: Exposes CPU, memory, disk, and other system statistics
+- **Container Health Monitoring**: Monitors container status and reports issues
+
+#### Security Design
+
+- **Local-only API**: HTTP API is only accessible within the bare metal machine, not publicly exposed
+- **Controlled Access**: Only communicates with the nilcc-agent on the same physical host
+- **Bootstrap Security**: Waits for authenticated bootstrap request before starting workloads
+
+#### Bootstrap Process
+
+The bootstrap process ensures secure workload initialization with proper credentials and configuration.
+
+**Initialization Phase**
+
+1. `cvm-agent` starts but waits for bootstrap request from `nilcc-agent`
+2. Bootstrap request includes:
+   - Docker Hub credentials (read-only access to avoid rate limits)
+   - ZeroSSL credentials for TLS certificate generation
+   - User-provided private registry credentials (if any)
+
+**Workload Deployment Phase**
+
+1. **Authentication**: Perform `docker login` with all provided credentials
+2. **Image Preparation**: Run `docker compose pull` to download all required images
+3. **Service Startup**: Execute `docker compose up` with:
+   - User's workload configuration
+   - Caddy proxy container (for TLS termination)
+   - nilcc-attester container (for attestation)
+4. **Certificate Monitoring**: Monitor Caddy for successful TLS certificate generation
+5. **Operational Mode**: Switch to log/stats serving once TLS is established
+
+This staged approach provides better error handling and ensures all components are properly configured before workload execution begins.
+
+### nilcc-api
+
+The [nilcc-api](https://github.com/NillionNetwork/nilcc/tree/main/nilcc-api) serves as the central control plane, providing the public interface for workload management and coordination across the entire nilCC infrastructure.
+
+#### Primary Functions
+
+- **User Interface**: Provides REST API endpoints for workload creation, management, and monitoring
+- **Agent Coordination**: Orchestrates multiple nilcc-agent instances across different bare metal hosts
+- **Event Processing**: Receives and processes status updates, errors, and events from agents
+- **Resource Allocation**: Makes decisions about workload placement based on available resources
+
+#### Integration Points
+
+- **Upward to Users/UI**: Serves the nilcc-ui dashboard and external API clients
+- **Downward to Agents**: Communicates with registered nilcc-agent instances using authentication tokens
+- **Data Persistence**: Maintains workload state, user accounts, and system configuration
```

Comment:
"Workload state" might be a confusing term since the workloads currently don't maintain state on reboot. Is this the workload configuration or workload logs?